api/agent/resource.go

```diff
@@ -284,26 +233,31 @@ func (a *resourceTracker) GetResourceToken(ctx context.Context, memory uint64, c
 	go func() {
 		defer span.End()
 		defer cancel()
+
+		var t ResourceToken
+		t = &resourceToken{err: CapacityFull}
```

Comment:
super nit: a comment here stating this is returned when ctx times out would be 🥇

Reply:
well, unless it can get one, of course... see previous comment for an idea about this :)